packages/app/components/Queue/TA/TAQueueListDetail.tsx

```diff
@@ -102,6 +103,27 @@ export default function TAQueueListDetail({
   const selectedQuestion = allQuestionsList.find(
     (q) => q.id === selectedQuestionId
   );
+  const navigateQuestions = (isUp: boolean) => {
+    const priorityAndWaitingQuestionIds = [
+      ...questions.priorityQueue,
+      ...questions.queue,
+    ].map((question) => question.id);
+    const numOfQuestions = priorityAndWaitingQuestionIds.length;
+    if (numOfQuestions > 0) {
+      setSelectedQuestionId((prevId) => {
+        const addMinus = isUp ? -1 : 1;
+        const qIdIndex = priorityAndWaitingQuestionIds.indexOf(prevId);
+        const modulusMagic =
+          (((qIdIndex + addMinus) % numOfQuestions) + numOfQuestions) %
+          numOfQuestions;
```

Comment:
obviously this works based off of my manual testing, but could you explain to me why you add `numOfQuestions` then mod by `numOfQuestions` again? i was trying to think about this and it seemed like just `((qIdIndex + addMinus) % numOfQuestions)` would work here.

Reply:
for the case when you press down many times until the number is decremented into a negative number. In which case, works funky in js/ts without adding the numquestions